### ISSUE TITLE
Simple login redirects don't work

### ISSUE THREAD

**Issue by peavers** [auth0-community/ember-simple-auth-auth0#100]
Not to be that guy raising a bunch of tickets but... 

```js
export default Route.extend(UnauthenticatedRouteMixin, {
  session: service(),

  actions: {
    login() {
      const lockOptions = {
        container: "login-container",
        oidcConformant: true,
        auth: {
          responseType: 'token',
          redirectUrl: http://localhost:4200/user,
          audience: "http://swordfish-service",
          params: {
            scope: 'openid profile'
          }
        }
      };
      get(this, 'session').authenticate('authenticator:auth0-lock', lockOptions);
    },
  }
})
```
I'd expect this to redirect to the user route when logging in. However a state mismatch error is thrown and the hash is left in the URL string, and the route remains `http://locahost:4200/login`. I am however authenticated, if I manually remove the hash from the URL, and change the route to where I want to go - I get there as the authenticated user. 

Setting `hashCleanup: true` seems to be ignored too. 

My config is very simple
```js
  ENV['ember-simple-auth'] = {
    authenticationRoute: 'login',
    auth0: {
      clientID: process.env.AUTH0_CLIENT_ID,
      domain: process.env.AUTH0_DOMAIN
    }
  };
```
Have I just missed the obvious?

**Comment by XaserAcheron** [comment]
No worries on the tickets -- bugs are bugs. Thanks for raisin' em!

I wonder if this is the same issue as #67 -- there's something funky about the URL hashes that's causing trouble. One user reported that manually removing the hash in the application route's `sessionAuthenticated` hook served as a workaround, and I'm curious if it would work for you too. If so, that's a pretty important clue regarding getting it fixed for real.

**Comment by peavers** [comment]
Ok, looks like the solve is simply including `email` in the scope, and setting `redirect` true. I've got my app redirecting as expected, and the URL being cleaned up. For anyone that finds this, 
```js
export default Route.extend(ApplicationRouteMixin, UnauthenticatedRouteMixin, {
  session: service(),

  actions: {
    login() {
      const lockOptions = {
        container: "login-container",
        oidcConformant: true,
        hashCleanup: true,
        auth: {
          redirect: true,
          audience: "http://swordfish-service",
          params: {
            scope: 'openid profile email',
          },
        }
      };

      get(this, 'session').authenticate('authenticator:auth0-lock', lockOptions);

    },

    logout() {
      get(this, 'session').invalidate();
    },
  }
})

```
And in my configuration
```
  ENV['ember-simple-auth'] = {
    authenticationRoute: '',
    routeAfterAuthentication: 'instances',
    routeIfAlreadyAuthenticated: 'instances',
    auth0: {
      clientID: process.env.AUTH0_CLIENT_ID,
      domain: process.env.AUTH0_DOMAIN
    }
  };
```

**Comment by XaserAcheron** [comment]
Glad this is working for you!

The URL hash shenanigans should be perma-fixed by #103 now, so hopefully folks won't stumble into this in the future.
